test/jb_nft_reward_datasource_delegate/did_pay.test.js

```diff
@@ -0,0 +1,191 @@
+  const ethToken = ethers.constants.AddressZero;
+
+  async function setup() {
+    let [deployer, projectTerminal, beneficiary, ...accounts] = await ethers.getSigners();
+
+    let [
+      mockJbDirectory,
+    ] = await Promise.all([
+      deployMockContract(deployer, jbDirectory.abi),
+    ]);
+
+    await mockJbDirectory.mock.isTerminalOf.withArgs(PROJECT_ID, projectTerminal.address).returns(true);
+    await mockJbDirectory.mock.isTerminalOf.withArgs(PROJECT_ID, beneficiary.address).returns(false);
+
+    const jbNFTRewardDataSourceFactory = await ethers.getContractFactory('JBNFTRewardDataSourceDelegate', deployer);
+    const jbNFTRewardDataSource = await jbNFTRewardDataSourceFactory
+      .connect(deployer)
+      .deploy(
+        PROJECT_ID,
+        mockJbDirectory.address,
+        1,
+        { token: ethToken, value: 1000000, decimals: 18, currency: CURRENCY_ETH },
+        NFT_NAME,
+        NFT_SYMBOL,
+        NFT_URI,
+        ethers.constants.AddressZero,
+        NFT_METADATA,
+        ethers.constants.AddressZero,
+      );
+
+    return {
+      projectTerminal,
+      beneficiary,
+      accounts,
+      jbNFTRewardDataSource,
+    };
+  }
+
+  it(`Should mint token if meeting contribution parameters`, async function () {
+    const { jbNFTRewardDataSource, projectTerminal, beneficiary } = await setup();
+
+    await expect(jbNFTRewardDataSource.connect(projectTerminal).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: ETH_TO_PAY, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    })).to.emit(jbNFTRewardDataSource, 'Transfer').withArgs(ethers.constants.AddressZero, beneficiary.address, 0);
+  });
+
+  it(`Should not mint token if exceeding max supply`, async function () {
+    const { jbNFTRewardDataSource, projectTerminal, beneficiary } = await setup();
+
+    await jbNFTRewardDataSource.connect(projectTerminal).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: ETH_TO_PAY, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    });
+
+    await expect(jbNFTRewardDataSource.connect(projectTerminal).didPay({
+      payer: beneficiary.address,
+      projectId: PROJECT_ID,
+      currentFundingCycleConfiguration: 0,
+      amount: { token: ethToken, value: ETH_TO_PAY, decimals: 18, currency: CURRENCY_ETH },
+      projectTokenCount: 0,
+      beneficiary: beneficiary.address,
+      preferClaimedTokens: true,
+      memo: '',
+      metadata: '0x42'
+    })).not.to.emit(jbNFTRewardDataSource, 'Transfer');
```

Comment:
That's what I meant in contract's comment -> no mint, no project token, just some eth gone :D